### PR TITLE
Add extension to measure bulk performance

### PR DESCRIPTION
**Pull request by nuzil**
### Description (*)
I created small extension for we can track performance of Asynchronous Import Module in future

Extension will add 2 new columns to magento_operation table with "created_at" and "finished_at" for we can track how much time each operation takes. 

In addition new endpoint were provided:

`GET {{URL}}rest/all/V1/bulk/:uuid/finish`

with response: 
```
[
    {
        "id": 115,
        "bulk_uuid": "80f38e7c-8470-4907-9766-0ec794d8b4f2",
        "topic_name": "async.magento.catalog.api.productrepositoryinterface.save.post",
        "serialized_data": null,
        "result_serialized_data": "}",
        "status": 1,
        "result_message": "Service execution success Magento\\Catalog\\Model\\ProductRepository\\Interceptor::save",
        "created_at": "2019-05-17 07:57:19",
        "finished_at": "2019-05-17 07:57:39",
        "error_code": null
    }
]
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/BulkPerformance/Api/Data/OperationInterface.php [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    public function getId();
+
+    /**
+     * Set operation id
+     *
+     * @param int $id
+     * @return $this
+     * @since 100.2.0
+     */
+    public function setId($id);
```
Comment (dmanners) [inline app/code/Magento/BulkPerformance/Api/Data/OperationInterface.php]: Do we cast types here in the methods for example `setId(int $id)` etc for the whole class.

File: app/code/Magento/BulkPerformance/Model/Operation.php [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    {
+        return $this->getData(self::STATUS);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setStatus($status)
+    {
+        return $this->setData(self::STATUS, $status);
```
Comment (dmanners) [inline app/code/Magento/BulkPerformance/Model/Operation.php]: General question on a design point of view, I know that there are a limited set of status options here. What are we thinking about validation against these items? At what point do we normally validate.

File: app/code/Magento/BulkPerformance/Model/OperationManagement.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        $resultData = null
+    ) {
+        try {
+            $operationEntity = $this->operationFactory->create();
+            $this->entityManager->load($operationEntity, $operationId);
+            $operationEntity->setErrorCode($errorCode);
+            $operationEntity->setStatus($status);
+            $operationEntity->setResultMessage($message);
+            $operationEntity->setSerializedData($data);
+            $operationEntity->setFinishedAt(date("Y-m-d H:i:s"));
```
Comment (dmanners) [inline app/code/Magento/BulkPerformance/Model/OperationManagement.php]: I am ok for using date here but I would to be consistent with other parts in Magento. What do we use for date processing in other options?

File: app/code/Magento/BulkPerformance/etc/db_schema.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Setup/Declaration/Schema/etc/schema.xsd">
+    <table name="magento_operation" resource="default" engine="innodb" comment="Operation entity">
+        <column xsi:type="timestamp" name="created_at" on_update="false" nullable="false" default="CURRENT_TIMESTAMP"
+                comment="Operation start time"/>
```
Comment (dmanners) [inline app/code/Magento/BulkPerformance/etc/db_schema.xml]: This comment is a little confusing for me. The column name is created at but the comment is around start time. I am not sure if created at will always be equal to start time or can you create and then start the operation later?

File: app/code/Magento/BulkPerformance/Model/Operation.php (hunk 2 above)
Comment (nuzil) [reply app/code/Magento/BulkPerformance/Model/Operation.php]: https://github.com/magento/async-import/issues/100

**Comment by nuzil** [comment]
We need a decisions: 
How we can deliver this performance required PR into Magento core. For other people can also test it

**Comment by kandy** [comment]
@nuzil Thanks for your contribution. 
Can I ask you compare this approach with Distributed tracing provided by most of APM (NewRelic, Datadog, Zipkin etc) solution?
Normal we don't recommend including any infrastructure related code in Core if it possible

**Comment by nuzil** [comment]
Hi @kandy 
Unfortunately for now its not suitable for us to use APMs we checked. Jmeter doing its job great here. 
I would not merge this code to core, I more looking for a solution to put it somewhere separately, or to some Performance repo, or making a new extension repo for such case

**Comment by nuzil** [comment]
Hey @dmanners I aligned code a bit. If all fine I would suggest to merge it so we can procceed and use if for performance measurements

**Comment by dmanners** [comment]
@nuzil I'm ok with this approach for now but I would love to see if we can do something with the performance testing tools instead.

**Pull request merged**